### DISCUSSION TITLE
Find out if a text is truncated.

### DISCUSSION THREAD

**Discussion by mutterer** [spakin/SimpInkScr#127] [Ideas]
Hi, 
When flowing text inside objects, it would be useful to be able to find out if the text is truncated. 
Here for example is where the text tool adjusts the indicator to red in the GUI whenever a text in truncated.
https://gitlab.com/inkscape/devel/inkscape/-/blob/master/src/ui/tools/text-tool.cpp#L551
Thanks for the super useful extension!

**Comment by spakin** [comment]
This indeed would be useful, but I don't believe truncation can be detected by extensions.  In general, information about text is available only to the Inkscape core, such as the file to which you linked above.  Extensions can't even acquire the bounding box of a piece of text!  (Simple Inkscape Scripting handles that one special case by spawning another copy of Inkscape and passing it the `--query-x --query-y --query-width --query-height` on the command line.)

If you or anyone else reading this issue knows of a mechanism for extensions to detect flowed-text truncation, please post it here.

**Comment by mutterer** [comment]
Thanks for the reply. FYI I looked into inkex issues and found this topic 🤞https://gitlab.com/inkscape/extensions/-/merge_requests/604

**Reply by spakin** [reply]
Nice.  Although that MR does not look like it will itself help with detecting truncated text in a frame, better inkex support for text is definitely a step in the right direction.
